### PR TITLE
ICMSLST-1270 - Exclusive Correspondence

### PR DESCRIPTION
**Pull request by marcuspp**
[](url)
Screenshot 2024-05-23 at 11 37 50

### CODE REVIEW

File: web/static/web/css/fox.css [hunk 1]
```diff
@@ -4764,6 +4764,10 @@ fieldset {
     background: #FFF5F5;
 }
 
+.hint-addon {
```
Comment (marcuspp) [inline web/static/web/css/fox.css:4767]: Adding this did seem to cause any issues elsewhere

File: web/domains/exporter/models.py [hunk 2]
```diff
@@ -55,6 +55,7 @@ class Exporter(Archivable, models.Model):
     main_exporter = models.ForeignKey(
         "self", on_delete=models.SET_NULL, blank=True, null=True, related_name="agents"
     )
+    exclusive_correspondence = models.BooleanField(null=False, default=False)
```
Comment (MattHolmes123) [inline web/domains/exporter/models.py:58]: same comment relating to null

File: web/mail/recipients.py [hunk 3]
```diff
@@ -35,17 +35,22 @@ def get_notification_emails(user: User) -> list[str]:
...
-    return get_organisation_contacts(org)
-
-
-def get_organisation_contacts(org: Organisation) -> QuerySet[User]:
-    obj_perms = get_org_obj_permissions(org)
-    return organisation_get_contacts(org, perms=[obj_perms.edit.codename])
+        exclusive_correspondence = application.exporter.exclusive_correspondence
+    contacts = get_organisation_contacts(org)
+    if exclusive_correspondence and application.contact in contacts:
+        return contacts.filter(pk=application.contact.pk)
```
Comment (MattHolmes123) [inline web/mail/recipients.py:52]: So if they are not in the list of contacts just default back to the org contacts?
Seems reasonable enough.

Do we care if the contact isn't in the list of contacts? 
e.g. should we log a sentry warning or something?

File: web/templates/web/domains/exporter/exclusive_correspondence_tooltip.html [hunk 4]
```diff
@@ -0,0 +1,3 @@
+<div id="correspondence-exclusive-tooltip" role="tooltip" class="hint-content" aria-hidden="true">
+  <div class="screen-reader-only">This item has hint text,</div>By default, correspondence about an application such as further information requests will be sent to all contacts below with the edit applications role. Tick this box so that correspondence is sent only to the individual contact selected on the application form.</div>
+</div>
```
Comment (MattHolmes123) [inline web/templates/web/domains/exporter/exclusive_correspondence_tooltip.html:3]: missing newline

**Review by mikeylockwood — APPROVED** [review]

**Pull request merged**